src/main/java/org/padaiyal/utilities/commandline/abstractions/OperatingSystem.java

```diff
@@ -0,0 +1,145 @@
+package org.padaiyal.utilities.commandline.abstractions;
+
+import java.util.Arrays;
+import java.util.Locale;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.padaiyal.utilities.I18nUtility;
+import org.padaiyal.utilities.PropertyUtility;
+
+/**
+ * Enum used to represent the operating system of a machine.
+ */
+public enum OperatingSystem {
+  WINDOWS(
+      "WINDOWS",
+      "CommandLineUtility.windows.where",
+      CommandLine.CMD,
+      CommandLine.POWERSHELL,
+      CommandLine.BASH,
+      CommandLine.ZSH,
+      CommandLine.SH
+  ),
+  LINUX(
+      "LINUX",
+      "CommandLineUtility.linux.which",
+      CommandLine.BASH,
+      CommandLine.ZSH,
+      CommandLine.SH
+  ),
+  @SuppressWarnings("SpellCheckingInspection")
+  MAC_OS_X(
+      "MAC OS X",
+      "CommandLineUtility.macos.which",
+      CommandLine.BASH,
+      CommandLine.ZSH,
+      CommandLine.TCSH,
+      CommandLine.KSH,
+      CommandLine.SH,
+      CommandLine.CSH,
+      CommandLine.CMD
+  ),
+  UNKNOWN(
+      "UNKNOWN",
+      null
+  );
+
+  /**
+   * Logger object.
+   */
+  private static final Logger logger = LogManager.getLogger(OperatingSystem.class);
+
+  static {
+    I18nUtility.addResourceBundle(
+        OperatingSystem.class,
+        OperatingSystem.class.getSimpleName() + "_i18n",
+        Locale.US
```

Comment:
why not add "_i18n"  to a property file if it is called in different classes? It will be easier to change if we ever decide to change "_i18n" to something else.

Reply:
I've something better. Removed the "i18n" prefix.